static/js/scheduler.js

```diff
@@ -0,0 +1,29 @@
+var scheduler = function(callback, time) {
+  this.callback = callback;
+  this.time = time;
+}
+
+scheduler.prototype.run = function(args, cb) {
+  var result = this.callback.apply(null, args);
+  delete this.timeoutID;
+  if (cb) cb(result);
+}
+
+scheduler.prototype.schedule = function(args, cb) {
+  this.cancel();
+  var self = this;
+  this.timeoutID = window.setTimeout(function() {
+    self.run(args, cb);
+  }, this.time, args, cb);
+}
+
+scheduler.prototype.cancel = function() {
+  if(typeof this.timeoutID == "number") {
```

Comment:
pq precisa verificar se é `number`? Não dá para simplesmente chamar a `clearTimeout(this.timeoutID);`?